### ISSUE TITLE
Maven NoClassDefFound Exceptions

### ISSUE THREAD

**Issue by etanshaul** [GoogleCloudPlatform/google-cloud-intellij#1939]
```
java.lang.NoClassDefFoundError: org/jetbrains/idea/maven/project/MavenProjectsManager
	at com.google.cloud.tools.intellij.appengine.java.project.DefaultAppEngineProjectService.isMavenModule(DefaultAppEngineProjectService.java:159)
	at com.google.cloud.tools.intellij.appengine.java.facet.standard.AppEngineStandardFacetEditor.isManagedLibrariesEnabled(AppEngineStandardFacetEditor.java:158)
	at com.google.cloud.tools.intellij.appengine.java.facet.standard.AppEngineStandardFacetEditor.createUIComponents(AppEngineStandardFacetEditor.java:149)
	at com.google.cloud.tools.intellij.appengine.java.facet.standard.AppEngineStandardFacetEditor.$$$setupUI$$$(AppEngineStandardFacetEditor.java)
	at com.google.cloud.tools.intellij.appengine.java.facet.standard.AppEngineStandardFacetEditor.<init>(AppEngineStandardFacetEditor.java:57)
	at com.google.cloud.tools.intellij.appengine.java.facet.standard.AppEngineStandardFacetConfiguration.createEditorTabs(AppEngineStandardFacetConfiguration.java:46)
	at com.intellij.facet.impl.ui.FacetEditorImpl.<init>(FacetEditorImpl.java:59)
	at com.intellij.facet.impl.ProjectFacetsConfigurator.getOrCreateEditor(ProjectFacetsConfigurator.java:189)
	at com.intellij.facet.impl.ui.facetType.FacetTypeEditor.a(FacetTypeEditor.java:86)
	at com.intellij.facet.impl.ui.facetType.FacetTypeEditor.createComponent(FacetTypeEditor.java:101)
```

```
Description
update failed for AnAction(com.google.cloud.tools.intellij.apis.AddCloudLibrariesAction) with ID=GoogleCloudTools.AddCloudLibraries: org/jetbrains/idea/maven/project/MavenProjectsManager
Stack trace:
java.lang.NoClassDefFoundError: org/jetbrains/idea/maven/project/MavenProjectsManager
	at com.google.cloud.tools.intellij.appengine.project.DefaultAppEngineProjectService.isMavenModule(DefaultAppEngineProjectService.java:159)
	at com.google.cloud.tools.intellij.apis.AddCloudLibrariesAction.checkModuleForAddCloudLibraries(AddCloudLibrariesAction.java:113)
	at java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:193)
	at java.util.Spliterators$ArraySpliterator.forEachRemaining(Spliterators.java:948)
	at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:481)
	at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:471)
	at java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:708)
	at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:499)
```

**Comment by etanshaul** [comment]
not reproducible. perhaps a classloading issue do to how the plugin was installed. closing

**Comment by etanshaul** [comment]
Turns out that this happens when the user uninstalls the Maven plugin (or sets up the IDE initially without it, if that's possible):

![image](https://user-images.githubusercontent.com/1735744/41168648-939fc5b6-6b14-11e8-8187-dc7c3388b386.png)

## Why did this happen pre-module-refactoring:

We had a hard depends on the maven plugin in our plugin.xml:

`<depends>org.jetbrains.idea.maven</depends>`

So the user should not have been prevented from uninstalling the Maven plugin and keeping the Cloud tools plugin installed at the same time.

The problem was that we also had an optional depends on the maven plugin defined above it:

`<depends optional="true" config-file="google-app-engine-maven-support.xml">org.jetbrains.idea.maven</depends>`

`plugin.xml`s have a quirk where you cannot depend on the same module twice. So essentially, only the optional depends was honored, hence allowing the user to uninstall the Maven plugin with no warning. 

Removing the optional depends, leads to the expected behavior:
![image](https://user-images.githubusercontent.com/1735744/41168922-50aa33bc-6b15-11e8-8444-f965e10a708f.png)

## Why does this happen now after the module refactoring?

It is the App Engine module that uses the maven integration. We don't have an explicit depends from the App Engine module to the maven module. Even if we did, the App Engine module itself is optional so having a hard dependency between App Engine and Maven would cause all of the App Engine features to stop working if the Maven plugin is uninstalled. 

The ideal situation is that if the user disables the Maven plugin, then _only_ Maven specific app engine features stop working.

**Comment by etanshaul** [comment]
fixed via #2192
